src/libfsm/minimise.c

```diff
@@ -7,49 +7,660 @@
 #include <assert.h>
 #include <stddef.h>
 #include <stdlib.h>
+#include <string.h>
+#include <stdint.h>
+
+#include <stdio.h>
 
 #include <fsm/fsm.h>
 #include <fsm/pred.h>
 #include <fsm/walk.h>
+#include <fsm/alloc.h>
 
+#include <adt/edgeset.h>
 #include <adt/set.h>
 
 #include "internal.h"
 
+#define LOG_MAPPINGS 0
+#define LOG_STEPS 0
+#define LOG_TIME 0
+#define LOG_INIT 0
+#define LOG_ECS 0
+#define LOG_PARTITIONS 0
+
+#if LOG_TIME
+#include <sys/time.h>
+#endif
+
+#include "minimise_internal.h"
+
 int
 fsm_minimise(struct fsm *fsm)
 {
-	int r;
+	int r = 0;
+	struct fsm *dst = NULL;
 
-	assert(fsm != NULL);
+	unsigned char labels[256];
```

Comment:
if this is for the number of symbols in the alphabet, there's a #define for this, `FSM_SIGMA_COUNT`
(and `FSM_SIGMA_MAX` for the maximum value of a symbol)

Reply:
Changed in 45a445c3.